llvm/lib/Target/AIE/AIEHazardRecognizerPRAS.cpp

```diff
@@ -24,7 +24,8 @@ using namespace llvm;
 
 AIEHazardRecognizerPRAS::AIEHazardRecognizerPRAS(const AIEBaseInstrInfo *TII,
                                                  const InstrItineraryData *II)
-    : AIEHazardRecognizer(TII, II, /*IsPreRA=*/false),
+    : AIEHazardRecognizer(TII, II, *(new AIEAlternateDescriptors()),
+                          /*IsPreRA=*/false),
```

Comment:
Maybe we could just add a `AIEAlternateDescriptors PRASAlternateDescriptors` attribute in `AIEHazardRecognizerPRAS`? Dynamic allocation is more error-prone than anything.